### PR TITLE
Don't remove + from file paths

### PR DESCRIPTION
**Pull request by keith**
Fixes https://github.com/prabirshrestha/vim-lsp/issues/167

### CODE REVIEW

**Comment by keith** [comment]
@prabirshrestha any thoughts on this one? 🙏

**Comment by mattn** [comment]
This is not an issue of vim-lsp. See https://github.com/prabirshrestha/vim-lsp/issues/167#issuecomment-450281985

**Comment by mattn** [comment]
Ah, sorry. This is my bad.

File: autoload/lsp/utils.vim [hunk 1]
```diff
@@ -5,8 +5,7 @@ endfunction
 " Decode uri function is taken from vital framework: https://github.com/vim-jp/vital.vim
 " For it's license (NYSL), see http://www.kmonos.net/nysl/index.en.html
```
Comment (mattn) [inline autoload/lsp/utils.vim]: Please remove this two lines of commet.

**Comment by mattn** [comment]
|case  |original path    |Language Server      |not patched      |patched          |
|------|-----------------|---------------------|-----------------|-----------------|
|clangd|`/path/to/C++.go`|`/path/to/C++.go`    |`/path/to/C  .go`|`/path/to/C++.go`|
|golsp |`/path/to/C++.go`|`/path/to/C%2b%2b.go`|`/path/to/C++.go`|`/path/to/C++.go`|
|?|`/path/to/C  .go`|`/path/to/C++.go`    |`/path/to/C++.go`|`/path/to/C++.go`|

This patch is okay to merge. But some language server may have this issue.

**Comment by prabirshrestha** [comment]
Specification has the code on how it should parse. https://microsoft.github.io/language-server-protocol/specification 

Uri is defined as https://tools.ietf.org/html/rfc3986

Reference implementation points to https://github.com/Microsoft/vscode-uri

**Comment by keith** [comment]
Does that mean I need to update something here to support that?

**Comment by prabirshrestha** [comment]
It will be good to try in vscode-uri package and see what it decodes too.

We now have a test infra so any code that touches this encoding should have tests. We have had several bugs reported before on it and updating that codepath without tests is now risky as it could cause other regressions.

**Comment by mattn** [comment]
Finally, I understand what is wrong.

```
function! s:decode_uri(uri) abort
    let l:ret = substitute(a:uri, '+', ' ', 'g')
    return substitute(l:ret, '%\(\x\x\)', '\=printf("%c", str2nr(submatch(1), 16))', 'g')
endfunction
```

The first substitute is wrong. Decoding `+` must be done for letters after `?`.

**Comment by mattn** [comment]
This is fixed by https://github.com/prabirshrestha/vim-lsp/pull/269

**Pull request closed without merging**